uadi.h

```diff
@@ -0,0 +1,63 @@
+#ifndef UADI_H
+#define UADI_H
+
+#include <stddef.h>
+
+#ifdef _WIN32
+#define DLL_EXPORT __declspec(dllexport)
+#else
+#define DLL_EXPORT
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef void* uadi_lib_handle;
+typedef void* uadi_device_handle;
+
+typedef unsigned char* uadi_chunk_ptr;
+
+typedef struct uadi_receive_struct{
+    uadi_chunk_ptr infopack_ptr;
+    uadi_chunk_ptr datapack_ptr;
+};
+
+// Error codes
+#define UADI_SUCCESS 0
```

Comment:
While macros have many known and unknown disadvantages, you still like to use them instead of an `enum`?

Reply:
Yeah, @odinthenerd and I discussed this earlier today. Even though the functionallity here matches the use-case of an `enum`, it is rarely seen in `DLL`-headers. 

Since we want the project to be used by as many people as possible, I would suggest to use the most idiomatic way and not the necessarily most correct way. Furthermore I am unsure how `FORTRAN` or `python` would handle this interface if it were an `enum`.

... but that's just my two cents